test/rules/do_not_create_setter_without_getter.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N do_not_create_setter_without_getter`
+
+class A {
+  // ignore: unused_field
+  int _x;
+
+  set x(int x) { // LINT
+    this._x = x;
```

Comment:
Not that it matters, but you could replace the whole class by
```
class A {
  set x(x) {} // LINT
}
```
which would shorten the test file. You can minimally remove the bodies from all of the members because this rule ignores them.